src/ucp/core/ucp_rkey.c

```diff
@@ -255,27 +255,32 @@ void ucp_rkey_dump_packed(const void *rkey_buffer, char *buffer, size_t max)
     ucp_md_map_t md_map;
     unsigned md_index;
     uint8_t md_size;
+    int first;
 
     snprintf(p, endp - p, "{");
     p += strlen(p);
 
     md_map = *(ucp_md_map_t*)(rkey_buffer);
     rkey_buffer += sizeof(ucp_md_map_t);
 
+    first = 1;
     ucs_for_each_bit(md_index, md_map) {
          md_size      = *((uint8_t*)rkey_buffer);
          rkey_buffer += sizeof(uint8_t);
 
+         if (!first) {
+             snprintf(p, endp - p, ",");
```

Comment:
why not ```p+= snprintf```?

Reply:
https://linux.die.net/man/3/snprintf
The functions snprintf() and vsnprintf() do not write more than size bytes (including the terminating null byte ('\0')). **If the output was truncated due to this limit then the return value is the number of characters (excluding the terminating null byte) which would have been written to the final string if enough space had been available.** Thus, a return value of size or more means that the output was truncated. (See also below under NOTES.)